packages/block-library/src/image/edit.js

```diff
@@ -104,9 +103,9 @@ export function ImageEdit( {
 	}, [ caption ] );
 
 	const ref = useRef();
-	const mediaUpload = useSelect( ( select ) => {
+	const { imageDefaultSize, mediaUpload } = useSelect( ( select ) => {
 		const { getSettings } = select( blockEditorStore );
-		return getSettings().mediaUpload;
+		return pick( getSettings(), [ 'imageDefaultSize', 'mediaUpload' ] );
 	} );
```

Comment:
While you are at this, it would be nice to fix this perf issue:
```suggestion
		return pick( getSettings(), [ 'imageDefaultSize', 'mediaUpload' ] );
	}, [] );
```

Similar work in the past https://github.com/WordPress/gutenberg/pull/23255 by @ZebulanStanphill.